### PR TITLE
2021 10 27 upgrade bitcoin s 1.8

### PR DESCRIPTION
**Pull request by Christewart**
Upgrade bitcoin-s to 1.8

As a by product of this PR I made some refactors to centralize logic around writing seeds on restoration/init flow. I also changed what returned by `showAndWait` from `DLCOracle` to `DLCOracleAppConfig`

I added compiler warnings over from the main bitcoin-s project to this project as well. There were errors uncovered and required fixing in this PR.

### CODE REVIEW

File: src/main/scala/com/krystal/bull/gui/landing/LandingPaneModel.scala [hunk 1]
```diff
@@ -71,11 +77,25 @@ class LandingPaneModel() {
...
+    }
+    val oracleAppConfigWithPw = DLCOracleAppConfig
+      .fromDatadir(KrystalBullAppConfig.DEFAULT_DATADIR, configOpt.toVector)
+
+    GlobalData.oracleAppConfigOpt = Some(oracleAppConfigWithPw)
+
+    val extKeyT =
+      Try(
+        WalletStorage.getPrivateKeyFromDisk(oracleAppConfigWithPw.seedPath,
+                                            SegWitTestNet3Priv,
```
Comment (Christewart) [inline src/main/scala/com/krystal/bull/gui/landing/LandingPaneModel.scala:96]: This is changed from `SegWitMainNetPriv` -> `SegWitTestNet3Priv`. Is this ok?

It needed to change because of this: 

https://github.com/bitcoin-s/bitcoin-s/blob/ee348c8a57691743de839b7696a5eb165909b124/dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala#L60

File: src/main/scala/com/krystal/bull/gui/config/KrystalBullAppConfig.scala [hunk 2]
```diff
@@ -85,7 +89,7 @@ object KrystalBullAppConfig {
                 "Roaming",
                 "KrystalBull")
     } else {
-      Paths.get(Properties.userHome, ".krystal-bull")
+      Paths.get(Properties.userHome, "krystal-bull")
```
Comment (benthecarman) [inline src/main/scala/com/krystal/bull/gui/config/KrystalBullAppConfig.scala:92]: need to revert this change

**Review by benthecarman — COMMENTED** [review]
Works for me now if you revert the change below

File: src/main/scala/com/krystal/bull/gui/config/KrystalBullAppConfig.scala (hunk 2 above)
Comment (Christewart) [reply src/main/scala/com/krystal/bull/gui/config/KrystalBullAppConfig.scala]: oops :man_facepalming:

**Review by benthecarman — APPROVED** [review]
